eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/JavadocCodeSnippetCheck.java

```diff
@@ -0,0 +1,252 @@
+        final DetailAST parametersToken = methodDefToken.findFirstToken(TokenTypes.PARAMETERS);
+        for (DetailAST ast = parametersToken.getFirstChild(); ast != null; ast = ast.getNextSibling()) {
+            if (ast.getType() != TokenTypes.PARAMETER_DEF) {
+                continue;
+            }
+
+            final DetailAST typeToken = ast.findFirstToken(TokenTypes.TYPE);
+            final DetailAST identToken = typeToken.findFirstToken(TokenTypes.IDENT);
+            String parameterType = "";
+            if (identToken != null) {
+                // For example, Map, String, Mono types
+                parameterType = identToken.getText();
+            } else {
+
+                DetailAST arrayDeclarator = typeToken.findFirstToken(TokenTypes.ARRAY_DECLARATOR);
+                if (arrayDeclarator == null) {
+                    // For example, int, boolean, byte primitive types
+                    parameterType = typeToken.getFirstChild().getText();
+                }
+
+                DetailAST arrayDeclaratorIterator = arrayDeclarator;
+                while (arrayDeclaratorIterator != null) {
+                    DetailAST temp = arrayDeclaratorIterator.findFirstToken(TokenTypes.ARRAY_DECLARATOR);
+                    if (temp == null) {
+                        // For example, int[][], byte[] types
+                        parameterType = arrayDeclaratorIterator.getFirstChild().getText();
+                        break;
+                    }
+                    arrayDeclaratorIterator = temp;
+                }
+            }
+            sb.append(parameterType).append("-");
+        }
+        int size = sb.length();
+        if (size == 0) {
+            return null;
+        }
+        return sb.substring(0, size - 1);
+    }
+
+    /**
+     *  Check if the given customDescription from codesnippet matched the naming pattern rules:
+     *  All rules now are case-insensitive.
+     *  <ol>
+     *    <li>Method names from codesnippet should have started with actual method name where the codesnippet javadoc represents</li>
+     *    <li>Parameters should match correctly in terms of ordering and naming.</li>
+     *  </ol>
+     *
+     * @param customDescription full sample code reference name from annotation codesnippet
+     * @param fullPath a string contains package name, class name, method name and parameters if exist.
+     * @param parameters parameters string which concatenate of argument types
+     * @return false if the given custom description not matched with naming rule. Otherwise, return true.
+     */
+    private boolean isNamingMatched(String customDescription, String fullPath, String parameters) {
+        String[] descriptionSegments = customDescription.split("#");
+        // One JavaDoc could have twe codesnippet. For example, for method name methodName(string, string),
+        // There are two codescippet, (1) packagename.classname.methodname#string-string
```

Comment:
```suggestion
        // There are two codesnippet, (1) packagename.classname.methodname#string-string
```

Reply:
You have marked this as resolved, but you don't appear to have fixed the typo I identified in this suggested change.

Reply:
I think I introduced the same typo again.